sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/StateStoreErrors.scala

```diff
@@ -173,8 +173,51 @@ object StateStoreErrors {
     StateStoreProviderDoesNotSupportFineGrainedReplay = {
     new StateStoreProviderDoesNotSupportFineGrainedReplay(inputClass)
   }
+
+
+  def invalidConfigChangedAfterRestart(configName: String, oldConfig: String, newConfig: String):
+  StateStoreInvalidConfigAfterRestart = {
+    new StateStoreInvalidConfigAfterRestart(configName, oldConfig, newConfig)
+  }
+
+  def duplicateStateVariableDefined(stateName: String):
+  StateStoreDuplicateStateVariableDefined = {
+    new StateStoreDuplicateStateVariableDefined(stateName)
+  }
+
+  def invalidVariableTypeChange(stateName: String, oldType: String, newType: String):
+  StateStoreInvalidVariableTypeChange = {
+    new StateStoreInvalidVariableTypeChange(stateName, oldType, newType)
+  }
 }
 
+class StateStoreDuplicateStateVariableDefined(stateName: String)
+  extends SparkRuntimeException(
+    errorClass = "STATE_STORE_DUPLICATE_STATE_VARIABLE_DEFINED",
```

Comment:
We had some error classes for transformWithState specifically right ? could we use that prefix here ?

Reply:
Done